python/ray/serve/controller.py

```diff
@@ -364,13 +364,16 @@ def deploy(
 
         autoscaling_config = deployment_config.autoscaling_config
         if autoscaling_config is not None:
-            previous_deployment = self.deployment_state_manager.get_deployment(name)
-            if previous_deployment is None:
-                deployment_config.num_replicas = autoscaling_config.min_replicas
+            if autoscaling_config.initial_replicas is not None:
+                deployment_config.num_replicas = autoscaling_config.initial_replicas
             else:
-                deployment_config.num_replicas = (
-                    previous_deployment.deployment_config.num_replicas
-                )
+                previous_deployment = self.deployment_state_manager.get_deployment(name)
```

Comment:
just want to call out: we don't inherit initial_replicas from the previous deployment?

Reply:
nice find of edge case here :D. Yeah looking at the logic if the previous deployment has initial_replicas, the new deployment will use num_replicas instead of initial_replicas. i think this is fine because the logic is used to make sure the # of replicas don't change within a given cluster during update

Reply:
Yeah, that's right. Unless the user explicitly sets `initial_replicas`, we use the previous `num_replicas` to prevent sudden downscaling/upscaling after an update.